File: clojure/leap/src/leap.clj
(ns leap)

(defn divisible-by
  [year number]
  (= 0 (rem year number)))

(defn leap-year?
  [year]  
  (and
    (divisible-by year 4)
    (or
      (not (divisible-by year 100))
      (divisible-by year 400))))

File: go/leap/leap.go
package leap

const testVersion = 3

// IsLeapYear returns true if given year is leap, false otherwise.
func IsLeapYear(y int) bool {
	if y%4 == 0 {
		if y%100 == 0 {
			return y%400 == 0
		}
		return true
	}
	return false
}
